### PR TITLE
circulation: block operations for expired patron

### PR DESCRIPTION
**Pull request by zannkukai**
* Checkout, request and renew circulation operations are blocked if the
  patron expiration date is reached.
* Introduces new utils fixtures and adapts unit tests.
* Closes rero/rero-ils#1495.

Co-Authored-by: Renaud Michotte <renaud.michotte@gmail.com>

## Code review check list

- [ ] Commit message template compliance.
- [ ] Commit message without typos.
- [ ] File names.
- [ ] Functions names.
- [ ] Functions docstrings.
- [ ] Unnecessary commited files?
- [ ] Cypress tests successful?

### CODE REVIEW

**Review by iGormilhit — APPROVED** [review]
Commit message approved.

**Review by benerken — APPROVED** [review]

File: rero_ils/modules/patrons/api.py [hunk 1]
```diff
@@ -434,9 +439,22 @@ def remove_role(self, role_name):
...
+        """Shortcut to find the patron expiration date."""
+        date_string = self.patron.get('expiration_date')
+        if date_string:
+            return datetime.strptime(date_string, '%Y-%m-%d')
+
+    @property
+    def is_expired(self):
+        """Check if patron expiration date is obsolete."""
+        expiration_date = self.expiration_date
+        return expiration_date and datetime.now() > expiration_date
```
Comment (rerowep) [inline rero_ils/modules/patrons/api.py:456]: Do we have to work with UTC times?

File: tests/fixtures/basics.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+@pytest.fixture(scope="module")
+def yesterday():
+    """Get yesterday timestamp."""
+    return datetime.now(timezone.utc) - timedelta(days=1)
+
+
+@pytest.fixture(scope="module")
+def tomorrow():
+    """Get tomorrow timestamp."""
+    return datetime.now(timezone.utc) + timedelta(days=1)
```
Comment (rerowep) [inline tests/fixtures/basics.py:34]: Shall we use `datetime.utcnow()` everywhere?

File: rero_ils/modules/patrons/api.py (hunk 1 above)
Comment (zannkukai) [reply rero_ils/modules/patrons/api.py:456]: `expiration_date` stored into the patron resource doesn't have any time (it's just YYYY-MM-DD).
Using `datetime.now()`, I'm sure tu use timestamp with server localized timezone.
So I think check is consistant and correct here

File: tests/fixtures/basics.py (hunk 2 above)
Comment (zannkukai) [reply tests/fixtures/basics.py:34]: This notation is a legacy from old unit test. See here : https://github.com/rero/rero-ils/blob/dev/tests/ui/holdings/test_serial_claims.py#L61

**Review by lauren-d — APPROVED** [review]

**Pull request merged**